docs: update injectable migration guide to reflect latest schematic changes

devversion opened the pull request:
Since we created the migration guide for the `missing-injectable` schematic, the schematic
changed in various ways. e.g. the migration no longer migrates classes passed to `useExisting`

Additionally the migration has been expanded to handle another Ivy breaking change where
providers like `{provide: X}` will be intepreted as `{provide: X, useClass: X}`. This pattern should
be documented in the migration guide.

devversion commented:
Initially I only wanted to add the new breaking change that will be handled, but I figured that we need to restructure things a bit since the guide is no longer only about _Missing injectable decorators_. Hope that is fine. I tried my best incorporating these two "schematic changes" into the guide.

mary-poppins commented:
You can preview cfac87e at https://pr33960-cfac87e.ngbuilds.io/.

mary-poppins commented:
You can preview e70b4ad at https://pr33960-e70b4ad.ngbuilds.io/.

kapunahelewong commented on aio/content/guide/migration-injectable.md:
```diff
@@ -71,15 +72,37 @@ to strictly require the decorator because doing so enables further
 adds any `@Injectable()` decorators that may be missing to future-proof your app.
 
+### Why is adding `useValue: undefined` necessary?
+
```
Maybe add an introductory sentence before the code snippet? Something like, "Consider the following pattern:"

You'd then probably want to change "Providers following this pattern..." to "Providers using this pattern..." in the next sentence just to avoid repetition.

devversion replied:
Great idea. Will do 👍

kapunahelewong commented on aio/content/guide/migration-injectable.md:
```diff
@@ -71,15 +72,37 @@ to strictly require the decorator because doing so enables further
+differently based on the Angular version in applications consuming your library.
 
-Yes, if your library has any tokens that are meant to be injected, they should be updated with the `@Injectable()` decorator. In a future version of Angular, a missing `@Injectable()` decorator will always throw an error.
+[DI_TOKEN]: guide/glossary#di-token
```
How about having the reference in a sentence like this:

> For more information, see the [DI token](url) entry in the glossary.

I'd write DI token like this (instead of "DI_TOKEN") because that's how it is in the glossary. If it should instead be "DI_TOKEN" then we should change it in the glossary to reflect this way of writing it. I vote for consistency in either case.

devversion replied:
This line is not display in the rendered page. It just serves a "variable" for the DI token glossary link. The "variable" will then be used in the page like followed: `The [DI token][DI_TOKEN] is used ...`

or do you mean something else?

kapunahelewong replied:
Oh! I see, never mind.

kapunahelewong reviewed (approved):
Very nice. Thank you. 😊I have only a couple of suggestions.

devversion committed:
docs: update injectable migration guide to reflect latest schematic changes

Since we created the migration guide for the `missing-injectable` schematic, the schematic
changed in various ways. e.g. the migration no longer migrates classes passed to `useExisting`

Additionally the migration has been expanded to handle another Ivy breaking change where
providers like `{provide: X}` will be intepreted as `{provide: X, useClass: X}`. This pattern should
be documented in the migration guide.

mary-poppins commented:
You can preview 2e1497d at https://pr33960-2e1497d.ngbuilds.io/.

IgorMinar reviewed (approved):
lgtm

IgorMinar commented:
this looks good to me, we need this info in the next RC, so I'll label it for merge.
